src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -535,14 +538,20 @@ private static boolean isThereActivityInCourse(Course course, ApiCallHandler api
 		return false;
 	}
 	
-	
+	/*
+	 * This is actually deleting the course , we have 'delete.course' properties from properties file if true then course will be deleted from canvas instance,
+	 * Set it to false for testing purposes
+	 */
 	private static void deleteTheCourse(Course course, ApiCallHandler apiHandler,CoursesForDelete coursesForDelete) {
-		HttpResponse httpResponse = apiHandler.getApiResponse(RequestTypeEnum.COURSE_DELETE, null, null, course.getCourseId());
-		httpResponseNullCheck(httpResponse,RequestTypeEnum.COURSE_DELETE);
-		int statusCode = httpResponse.getStatusLine().getStatusCode();
-		if(statusCode!=200) {
-			M_log.error(apiCallErrorHandler(httpResponse,RequestTypeEnum.COURSE_DELETE,apiHandler));
-			return;
+		if(isCourseDeleteEnabled) {
+			M_log.debug("courses enabled for delete");
+			HttpResponse httpResponse = apiHandler.getApiResponse(RequestTypeEnum.COURSE_DELETE, null, null, course.getCourseId());
+			httpResponseNullCheck(httpResponse,RequestTypeEnum.COURSE_DELETE);
+			int statusCode = httpResponse.getStatusLine().getStatusCode();
+			if(statusCode!=200) {
+				M_log.error(apiCallErrorHandler(httpResponse,RequestTypeEnum.COURSE_DELETE,apiHandler));
+				return;
+			}
 		}
 		M_log.info("***** Course Deleted: "+course.getCourseId());	
```

Comment:
It would be helpful (and reassuring to those running this application) if this log message included some indication that the deletion didn't actually happen.  While we do have the "courses enabled for delete" message above, it's probably easy for one to focus on these "Course Deleted" messages and have an unnecessary panic attack.  Changing this log message to "Course Deleted (skipped)" when deletion isn't enabled would do the job.